vizro-core/src/vizro/models/_components/button.py

```diff
@@ -24,11 +25,17 @@ class Button(VizroBaseModel):
 
     type: Literal["button"] = "button"
     text: str = Field("Click me!", description="Text to be displayed on button.")
+    href: str = Field("", description="URL (relative or absolute) to navigate to.")
     actions: list[Action] = []
 
     # Re-used validators
     _set_actions = _action_validator_factory("n_clicks")
 
     @_log_call
     def build(self):
-        return dbc.Button(id=self.id, children=self.text)
+        return dbc.Button(
+            id=self.id,
+            children=self.text,
+            href=get_relative_path(self.href) if self.href.startswith("/") else self.href,
+            target="_top",
```

Comment:
I was surprised to see that we set `target="_top"` here (as in `vm.Card`) and had to look back at https://github.com/mckinsey/vizro/pull/585 to remind myself why.

Please could you just check this works as expected with _relative_ links though? If you have a dashboard in pycafe with a button and card pointing to another page in the same dashboard, it should not break out of the iframe but instead stay in it. It's only external links that should break out of the iframe. I think this probably works correctly already but would be good to check. If it's not working already then we should only set `target="_top"` conditionally for external links only.

Reply:
Yes, that seems to work as intended 👍 https://py.cafe/huong-li-nguyen/vizro-button-link-examples